apps/builder/app/builder/features/style-panel/shared/css-value-input/css-value-input.stories.tsx

```diff
@@ -1,15 +1,91 @@
 import * as React from "react";
 import { Flex, DeprecatedTextField } from "@webstudio-is/design-system";
-import type { StyleValue } from "@webstudio-is/css-engine";
+import type {
+  InvalidValue,
+  StyleProperty,
+  StyleValue,
+} from "@webstudio-is/css-engine";
 import { CssValueInput, type IntermediateStyleValue } from "./css-value-input";
 import { action } from "@storybook/addon-actions";
 import { toValue } from "@webstudio-is/css-engine";
 import { theme } from "@webstudio-is/design-system";
+import { isValidDeclaration, parseCssValue } from "@webstudio-is/css-data";
 
 export default {
   component: CssValueInput,
 };
 
+export const WithCustomValidator = () => {
+  const [value, setValue] = React.useState<StyleValue>({
+    type: "unit",
+    value: 0,
+    unit: "px",
+  });
+
+  const [intermediateValue, setIntermediateValue] = React.useState<
+    StyleValue | IntermediateStyleValue
+  >();
+
+  return (
+    <CssValueInput
+      styleSource="preset"
+      property="boxShadow"
+      value={value}
+      intermediateValue={intermediateValue}
+      keywords={[]}
+      onChange={(value) => {
+        setIntermediateValue(value);
+      }}
+      onHighlight={(value) => {
+        action("onHighlight")(value);
+      }}
+      onChangeComplete={({ value }) => {
+        // on blur, select, enter etc.
+        setValue(value);
+        setIntermediateValue(undefined);
+        action("onChangeComplete")(value);
+      }}
+      onAbort={() => {
+        action("onAbort")();
+      }}
+      isValidDeclaration={(_, value) =>
+        isValidDeclaration("boxShadow", `${value} 0`)
+      }
+      parseIntermediateOrInvalidValue={(
+        _: StyleProperty,
+        styleValue: IntermediateStyleValue | InvalidValue
+      ): StyleValue => {
+        const value = styleValue.value.trim();
+        const unit = "unit" in styleValue ? styleValue.unit ?? "px" : "px";
+
+        /* Box shaodw needs atleast two values, so we are passing custom validator */
+        let styleInput = parseCssValue("boxShadow", `${value}${unit} 0`);
```

Comment:
can't we just check any unit value here like `width` using default parseBlabla... ?

Reply:
And if yes, we can just pass fake property name instead of all the code above ? 
No need in abstraction like above in a single edge case

Reply:
That's actually the first idea in my mind, but i was just not sure if passing a `fake` property with same defaults is allowed 😄. If allowed, yeah that's the easiest, cleanest approach 🚀